pkg/operation/botanist/component/resourcemanager/resource_manager.go

```diff
@@ -218,27 +220,34 @@ func (r *resourceManager) Deploy(ctx context.Context) error {
 		return fmt.Errorf("missing server secret information")
 	}
 
+	var fns []func(context.Context) error
 	if r.values.TargetDiffersFromSourceCluster {
 		r.secrets.shootAccess = r.newShootAccessSecret()
 		if err := r.secrets.shootAccess.WithTokenExpirationDuration("24h").Reconcile(ctx, r.client); err != nil {
 			return err
 		}
-	}
-
-	fns := []func(context.Context) error{
-		r.ensureServiceAccount,
-		r.ensureRBAC,
-		r.ensureService,
-		r.ensureDeployment,
-		r.ensurePodDisruptionBudget,
-		r.ensureVPA,
-	}
 
-	if r.values.TargetDiffersFromSourceCluster {
-		fns = append(fns, r.ensureShootResources)
-		fns = append(fns, r.ensureNetworkPolicy)
+		fns = []func(context.Context) error{
+			r.ensureServiceAccount,
+			r.ensureRBAC,
+			r.ensureService,
+			r.ensureDeployment,
+			r.ensurePodDisruptionBudget,
+			r.ensureVPA,
+			r.ensureShootResources,
+			r.ensureNetworkPolicy,
+		}
 	} else {
-		fns = append(fns, r.ensureMutatingWebhookConfiguration)
+		fns = []func(context.Context) error{
+			r.ensureServiceAccount,
+			r.ensureRBAC,
+			r.ensurePriorityClass,
```

Comment:
Why is the `PriorityClass` only used in this case?

Reply:
I think we should have a separate issue for the PriorityClasses for the Shoot control plane. From what I recall only few components in the control plane had PriorityClasses. I want to say that it would be not okay to have a PriorityClass for GRM while we don't have for much important components like kube-scheduler, kube-controller-manager, etc. I can try to open at least issue for it.